src/commonMain/kotlin/exchange.dydx.abacus/processor/router/skip/SkipStatusProcessor.kt

```diff
@@ -0,0 +1,179 @@
+package exchange.dydx.abacus.processor.router.skip
+
+import exchange.dydx.abacus.processor.base.BaseProcessor
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.utils.mutable
+
+internal class SkipStatusProcessor(
+    parser: ParserProtocol,
+    private val transactionId: String?,
+) : BaseProcessor(parser) {
+
+    enum class TransferTypes(val value: String) {
+        IBCTransfer("ibc_transfer"),
+        AxelarTransfer("axelar_transfer"),
+        CCTPTransfer("cctp_transfer")
+    }
+
+    enum class TransferDirection(val value: String) {
+        From("from"),
+        To("to"),
+    }
+
+    data class ChainData(
+        val chainId: String? = null,
+        val transactionUrl: String? = null,
+        val txHash: String? = null,
+        val status: String? = null
+    ) {
+        fun toRouteStatusMap(): Map<String, String?> {
+            return mapOf(
+                "chainId" to chainId,
+                "txHash" to txHash,
+                "status" to status,
+            )
+        }
+        fun toTransferChainStatusMap(): Map<String, String?> {
+            return mapOf(
+                "transactionId" to txHash,
+                "transactionUrl" to transactionUrl,
+            )
+        }
+    }
+
+    private fun getStrFromMap(data: Map<String, Any?>, path: String): String? {
+        return parser.asString(parser.value(data, path))
+    }
+
+    private fun getSquidStatusFromState(state: String?): String? {
+        if (state == null) return null
+        if (state.contains("SUCCESS") || state.contains("RECEIVED")) return "success"
+        return "ongoing"
+    }
+
+    private fun getStatusFromTransferState(state: String?, transferDirection: TransferDirection): String? {
+//        CCTP transfer state:
```

Comment:
nit multiline comments are possible via /** **/

Reply:
yah my b, got lazy with the copypasta tbh